sources/bootconfig/src/ser.rs

```diff
@@ -0,0 +1,543 @@
+//! The serialization module implements serialization support for kernel boot config files
+//! A lot of the logic is borrowed from Bottlerocket datastore serialization since the data
+//! format for kernel boot config files is very similar
+pub use crate::error;
+pub use crate::error::{Error, Result};
+use log::trace;
+use serde::{ser, Serialize};
+use snafu::OptionExt;
+
+/// Primary interface for serializing a kernel boot config out to string
+pub fn to_string<T>(value: &T) -> Result<String>
+where
+    T: ?Sized + Serialize,
+{
+    // Preallocate string buffer
+    let mut output = String::with_capacity(128);
```

Comment:
Curious how you chose the starting capacity. I think it makes sense, but I went down the rabbit hole of trying to find docs on how Rust chooses a default capacity and couldn't find what I was looking for!

Reply:
It's mostly arbitrary. I took it from serde-json and toml-rs (e.g. [toml_rs](https://github.com/alexcrichton/toml-rs/blob/70caf4085324bff452d8c20ead55974513030696/src/ser.rs#L91)).  It's hard to come up with a heuristic for guesstimating how much we should preallocate.

Reply:
As a followup: I spent some more time peeking at the implementation of `Vec` and I believe by default `capacity` is set to 0 to delay allocations in cases where none occur.

Anyways, I totally buy in to the idea of having something preallocated here.